src/vsr/superblock_free_set.zig

```diff
@@ -448,6 +448,14 @@ pub const FreeSet = struct {
         // Index verification is O(blocks.bit_length) so do it only at checkpoint, which is
         // also linear.
         set.verify_index();
+
+        var address_previous: u64 = 0;
+        for (free_set_encoded_blocks) |address| {
+            assert(address > 0);
+            assert(address > address_previous);
+            address_previous = address;
+            set.release(address);
+        }
```

Comment:
Ah, so we can't use `.mark_released()` here because, unlike in `open()`, the bits are already set in `blocks`.

Reply:
Yeah, I also don't particularly like this semi-duplication between `open -> mark_release` and `checkpoint`, but I couldn't find an obvious way to eliminate this. 

The current implementation also relies heavily on `include_staging / exuclude_staging` hack, which I am also not a fan of.

_But_ I actually like how we atomically checkpoint free set at the end in this PR, such that all prior grid operations (repairs&creation of free set own blocks) operate in the old checkpoint.